source/sdk/threading/native/ThreadUnix.ooc

```diff
@@ -76,24 +63,17 @@ version(unix || apple) {
 			ts tv_sec = floor(absSeconds) as TimeT
 			ts tv_nsec = ((absSeconds - ts tv_sec) * 1000 + 0.5) * (1_000_000 as Long)
 		}
-
-		/*
-		 * This is a semi-busywait implementation of timedjoin
-		 * The real solution is to use condition variables, but
-		 * I honestly don't have the courage to use them right now.
-		 */
 		__fake_timedjoin: func (seconds: Double) -> Bool {
-			while (seconds > 0.0) {
+			//TODO: This *is* fake, use WaitCondition instead
+			status := false
+			while (seconds > 0.0 && !status) {
 				Time sleepMilli(20)
 				seconds -= 0.02
-				if (!alive?()) {
-					// successfully joined
-					return true
-				}
+				if (!alive?())
+					status = true
 			}
-			false
+			status
```

Comment:
Same here - you added `status`, might as well call it `result` and get it right the first time.